### ISSUE TITLE
[bug] `Version` and `location_lat` and `location_lng` not being tracked

### ISSUE THREAD

**Issue by BrodaNoel** [amplitude/Amplitude-TypeScript#179]
After migrating to this new library, I'm seeing events without `Version`, which was automatically tracked before (`expo-analytics-amplitude` who was a wrapper of `Amplitude-Android` and `Amplitude-iOS`).

(I'm also seeing `null` the `location_lat` and `location_lng`, which was automatically tracked before as well)

As you can see, there is a `None` now:

<img width="1161" alt="Screen Shot 2022-08-10 at 15 09 05" src="https://user-images.githubusercontent.com/1837969/183985606-7be3bef3-814e-489e-9953-3945d347f6ef.png">

<img width="1132" alt="Screen Shot 2022-08-10 at 15 09 23" src="https://user-images.githubusercontent.com/1837969/183985651-5af68fdc-02c3-40f9-9135-c7417fa34c78.png">

Raw:
```
{
  "$insert_id": "f82b7e89-3e36-4c0b-b508-00c21d09ee74",
  "$insert_key": "010cf28863bc2a1de01d0666c5da3fe85f#208",
  "$schema": 13,
  "_time": 1659654506324,
  "adid": null,
  "amplitude_attribution_ids": null,
  "amplitude_event_type": null,
  "amplitude_id": 441735958251,
  "app": 227935,
  "city": null,
  "client_event_time": "2022-08-04 23:08:26.324000",
  "client_upload_time": "2022-08-04 23:08:29.271000",
  "country": "United States",
  "data": {
    "group_first_event": {
    },
    "group_ids": {
    }
  },
  "data_type": "event",
  "device_brand": null,
  "device_carrier": "Cell-c",
  "device_family": "Google Nexus Phone",
  "device_id": "ae40a07e-8a16-4a0d-96c4-beb80f4960a2",
  "device_manufacturer": "LGE",
  "device_model": "Nexus 5X",
  "device_type": "Google Nexus 5X",
  "display_name": "Go to Explore",
  "dma": null,
  "event_id": 0,
  "event_properties": {
  },
  "event_time": "2022-08-04 23:08:26.324000",
  "event_type": "PAGE_EXPLORE",
  "global_user_properties": {
  },
  "group_properties": {
  },
  "groups": {
  },
  "idfa": null,
  "ip_address": "xxx",
  "is_attribution_event": false,
  "language": "English",
  "library": "amplitude-react-native-ts/0.2.1",
  "location_lat": null,
  "location_lng": null,
  "os": "android 6.0.1",
  "os_name": "android",
  "os_version": "6.0.1",
  "partner_id": null,
  "paying": null,
  "plan": {
  },
  "platform": "android",
  "region": null,
  "sample_rate": null,
  "server_received_time": "2022-08-04 23:08:29.271000",
  "server_upload_time": "2022-08-04 23:08:29.272000",
  "session_id": 1659654506118,
  "source_id": null,
  "start_version": null,
  "timeline_hidden": false,
  "user_creation_time": "2022-08-04 23:08:26.324000",
  "user_id": null,
  "user_properties": {
  },
  "uuid": "xxx",
  "version_name": null
}
```

**Comment by liuyang1520** [comment]
@BrodaNoel , thanks for reporting this! I will take a look at this.

**Comment by BrodaNoel** [comment]
`Carrier` (Amplitude native value) not being tracked either

**Comment by BrodaNoel** [comment]
@kevinpagtakhan why is this consider an Enhancement and not a bug? `Version` is a native value from Amplitude. Without it, it's impossible to use the charts related to the app version (or even the whole "Releases" dashboard)

**Comment by ahilles107** [comment]
here is my fix for missing Version property:

```
amplitudeInit("xxxxxx", undefined, {
  // @ts-ignore
  versionName: true,
  appVersion: Application.nativeApplicationVersion?.toString() ?? "",
});
```

I'm not sure which config property fixes that, will check that on Monday.

**Comment by BrodaNoel** [comment]
I just realized that this bug is a bit more important as I was expecting at the beginning.
[Some days ago I saw that the `init` function was duplicating users if called more than 1 time](https://github.com/amplitude/Amplitude-TypeScript/issues/178), which I "fixed" calling the `init` on the app startup.
The problem is that as you can see in the following chart, I'm still seeing A LOT of "New User" events, even considering that now I'm calling `init` and theeeen I track the other events.

<img width="1134" alt="Screen Shot 2022-08-16 at 14 41 51" src="https://user-images.githubusercontent.com/1837969/184944245-1ead8f00-511f-47d7-adc3-f380561fddd9.png">

So, what it comes to my mind is this:

I implemented this new library in my app version `14.0.0`, and I fixed the `init` call in version `14.2.0`, so maaaaybe those New Users events that still being wrongly tracked are users that still using app version `14.0` or `14.1`...
The problem is that I can't know because Amplitude is not tracking version...
I still see a lot of "New Users" that should not be there, that maaaybe they are in `14.2` (which could imply that I actually didn't fix the problem), but I won't know it until Amplitude track my app version properly again.

**Comment by kevinpagtakhan** [comment]
Hi @BrodaNoel, I had tried calling `init` multiple times but was unable to repro the "duplicated user" issue but I was testing on react native web, perhaps you were testing using mobile? How it ends up for me is when init is called multiple times, it sends events with the same device id which resolves to the same user on amplitude.

Regarding app version, I'm trying to help figure out what the issue is since I'm seeing code paths that tell me we should be tracking the app version so I must be missing something. In the meantime, if you have the app version handy in your JS/TS code, you can also pass it to config and we'll track using that

```ts
amplitude.init(API_KEY, undefined, {
  appVersion: 'asdf',
});
```

**Comment by BrodaNoel** [comment]
@kevinpagtakhan yes I am using mobile.
If you check the screenshots in my main report (https://github.com/amplitude/Amplitude-TypeScript/issues/178), the Device ID change on every new init. That may be the issue behind the duplicated users.

**Comment by BrodaNoel** [comment]
BTW, thanks for the `appVersion` point. I will implement it right now and then I'll create a new issue in case the duplication still present. The app version will help me to understand what is going on there.

**Comment by kevinpagtakhan** [comment]
@BrodaNoel yup I remember seeing different device IDs, which seems to be the issue. It's why I think all the `init` calls happen too fast and they all try to read from an empty storage then try to generate a new device ID on each `init` call. Ideally, the succeeding `init` should read the device ID used by the previous ones.

**Comment by BrodaNoel** [comment]
@kevinpagtakhan I'm gonna be checking how that works.
I just released a new version of my app, with the `appVersion` fixed.

After some users start using it, I'll check what is going on there. Thanks!

**Comment by BrodaNoel** [comment]
The workaround works properly!

```
import Constants from 'expo-constants';

await initAmplitude(apiKey, undefined, {
      appVersion: Constants.manifest.version,
      trackingOptions: {
        adid: false,
      },
    }).promise;
```

**Comment by bohan-amplitude** [comment]
Is this issue resolved @kevinpagtakhan @liuyang1520 @BrodaNoel

**Comment by BrodaNoel** [comment]
As far as I understand, no. It should be tracking automatically the app version. And currently it only works if I explicitly set the version.

Let me know if you release a new version, so I can test it and report here the progress

**Comment by daveslutzkin** [comment]
Any update on this? We just moved to the new SDK and the app version isn't being tracked.

**Comment by Gedzis** [comment]
Any update on this? Do you have timeline when it gona be fixed? We are planing upgrading SDK version, but this stops us from doing this.

**Comment by BrodaNoel** [comment]
I just checked. This still not workingBroda NoelEl 30 ene. 2023, a la(s) 05:39, Gedzis ***@***.***> escribió:﻿
Any update on this? Do you have timeline when it gona be fixed? We are planing upgrading SDK version, but this stops us from doing this.

**Comment by daveslutzkin** [comment]
Yep still broken here too

**Comment by bsonmez** [comment]
There is still an issue with the app version. It doesn't send the latest version numbers.

**Comment by justin-fiedler** [comment]
Hi everyone,

`appVersion` support was fixed in [v1.3.2](https://github.com/amplitude/Amplitude-TypeScript/releases/tag/%40amplitude%2Fanalytics-react-native%401.3.2).

To remove permissions from the SDK that may not be required for all customers, we extracted location support from the core SDK. To use location you will now need to create an enrichment [Plugin](https://www.docs.developers.amplitude.com/data/sdk-plugins/) to set the location yourself. 

Here is an [example of how to set `location_lat` and `location_lng`](https://github.com/amplitude/Amplitude-TypeScript/blob/1.x/examples/plugins/react-native-get-location-plugin/LocationPlugin.ts) in [v1.3.6](https://github.com/amplitude/Amplitude-TypeScript/releases/tag/%40amplitude%2Fanalytics-react-native%401.3.6).

Carrier support should work on Android, but Apple stopped supporting this in iOS 16. Before we [fetched carrier info](https://github.com/amplitude/Amplitude-iOS/blob/94b4dcfda9ed9edd6463d9a657bf30c9cf075f09/Sources/Amplitude/AMPDeviceInfo.m#L121)by using serviceSubscriberCellularProviders and CTCarrier which are [deprecated ](https://developer.apple.com/documentation/coretelephony/cttelephonynetworkinfo/3024511-servicesubscribercellularprovide)with [no replacement](https://developer.apple.com/forums/thread/714876?answerId=728276022#728276022). If you have recommendations on how to better support carrier on Apple let us know. You can also set it yourself using a Plugin.

Hope this helps!
